pkg/prometheus/agent/statefulset.go

```diff
@@ -391,7 +396,7 @@ func makeStatefulSetSpec(
 				Volumes:                       volumes,
 				Tolerations:                   cpf.Tolerations,
 				Affinity:                      cpf.Affinity,
-				TopologySpreadConstraints:     cpf.TopologySpreadConstraints,
+				TopologySpreadConstraints:     tscs,
```

Comment:
(nit) we could factorize code a bit:
```suggestion
				TopologySpreadConstraints:     prompkg.MakeK8sTopologySpreadConstraints(finalSelectorLabels, cpf.TopologySpreadConstraints)),
```